packages/sdk-react/src/components/receivables/InvoiceDetails/CreateReceivable/sections/components/ReminderSection/RemindersSection.tsx

```diff
@@ -0,0 +1,189 @@
+import { useState } from 'react';
+import {
+  Controller,
+  ControllerRenderProps,
+  FieldValues,
+  useFormContext,
+} from 'react-hook-form';
+
+import { CreateReceivablesFormProps } from '@/components/receivables/InvoiceDetails/CreateReceivable/validation';
+import { useMoniteContext } from '@/core/context/MoniteContext';
+import { useRootElements } from '@/core/context/RootElementsProvider';
+import { t } from '@lingui/macro';
+import { useLingui } from '@lingui/react';
+import {
+  Alert,
+  Box,
+  Card,
+  CardContent,
+  Grid,
+  SelectChangeEvent,
+  Stack,
+  Typography,
+} from '@mui/material';
+
+import type { SectionGeneralProps } from '../../Section.types';
+import { useOverdueReminderById } from './hooks/useOverdueReminderById';
+import { usePaymentReminderById } from './hooks/usePaymentReminderById';
+import { useReminderPermissions } from './hooks/useReminderPermissions';
+import { useValidateCounterpart } from './hooks/useValidateCounterpart';
+import { SelectFieldWithEdit } from './SelectFieldWithEdit';
+
+export const ReminderSection = ({ disabled }: SectionGeneralProps) => {
+  const { i18n } = useLingui();
+  const { control } = useFormContext<CreateReceivablesFormProps>();
+  const { root } = useRootElements();
+  const { api } = useMoniteContext();
+
+  const {
+    isReadPaymentReminderAllowedLoading,
+    isReadOverdueReminderAllowedLoading,
+  } = useReminderPermissions();
+
+  const [
+    selectedPaymentIDReminderDetails,
+    setSelectedPaymentIDReminderDetails,
+  ] = useState<string | undefined>(undefined);
+
+  const [
+    selectedOverdueIDReminderDetails,
+    setSelectedOverdueIDReminderDetails,
+  ] = useState<string | undefined>(undefined);
+
+  const { isEmailValid, areRemindersEnabled } = useValidateCounterpart();
+
+  const { data: paymentReminders, isLoading: isPaymentRemindersLoading } =
+    api.paymentReminders.getPaymentReminders.useQuery({});
+  const { data: overdueReminders, isLoading: isOverdueRemindersLoading } =
+    api.overdueReminders.getOverdueReminders.useQuery({});
+
+  const { data: paymentIDReminder } = usePaymentReminderById(
```

Comment:
Since you are just the only one who needs these API-hooks just one time
* I see no reason to export them
* To create them if the hook is used just once

I would just not add any extra responsibility to the code. An extra file, an extra exported function - is an extra responsibility.

Reply:
Done, changed logic